BinaryEncoding.md

```diff
@@ -201,9 +199,13 @@ invalidate a module.
 | name | `bytes` ? | section name string, present if `id == 0` |
 | payload_data  | `bytes` | content of this section, of length `payload_len - sizeof(name) - sizeof(name_len)` |
 
-Each known section is optional and may appear at most once.
-Custom sections all have the same `id`, and can be named non-uniquely (all bytes composing their names can be identical).
-Known sections from this list may not appear out of order.
+Each known section is optional and may appear at most once. Custom sections all have the same `id` (0), and can be named non-uniquely (all bytes composing their names may be identical).
+
+Custom sections are intended to be used for debugging information, future evolution, or third party extensions. For MVP, we use a specific custom section (the [Name Section](#name-section)) for debugging information. 
+
+If a wasm implementation reads and validates any custom section during module validation time, such validation errors do not invalidate the module.
```

Comment:
This isn't part of validation, so I wouldn't call it that. How about:

"If a WebAssembly implementation interprets the payload of any custom section during module validation or compilation, errors in that payload must not invalidate the module."